src/mlpack/core/math/digamma.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file core/math/digamma.hpp
+ * @author Gopi Tatiraju 
+ *
+ * Some parts of the implementation are inspired from boost.
+ * Boost Software License, Version 1.0. (See accompanying file
+ * LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_MATH_DIGAMMA_HPP
+#define MLPACK_CORE_MATH_DIGAMMA_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace math {
+
+/**
+ * This fucntion calculates and returns digamma(x).
```

Comment:
@heisenbuug change `fucntion` to `function`  
seems to be an often made spelling mistake  
https://www.dictionary.com/browse/function

Reply:
Yup, I know the correct spelling. But somehow it feels like my fingers are programmed to make the mistake.
No excuses :sweat_smile: :sweat_smile: